core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/neo/BatchToTupleConverter.scala

```diff
@@ -1,13 +1,20 @@
 package edu.uci.ics.amber.engine.architecture.worker.neo
 
+import edu.uci.ics.amber.engine.architecture.worker.neo.WorkerInternalQueue.{
+  DataBatch,
+  DataPayload,
+  DummyPayload,
+  EndPayload
+}
 import edu.uci.ics.amber.engine.common.InputExhausted
 import edu.uci.ics.amber.engine.common.tuple.ITuple
 
 class BatchToTupleConverter(internalQueue: WorkerInternalQueue) {
 
   // save current batch related information
-  private var currentBatch: (Int, Array[ITuple]) = _
-  private var currentTupleIndex = 0
+  private var currentBatch: DataBatch = _
+  private var currentInput = 0
```

Comment:
`currentInputTuple` ?

Reply:
not clear on what the variable represents from its name, also please add a short comment to explain it